src/SourcePathsMapping/include/SourcePathsMapping/MappingManager.h

```diff
@@ -21,6 +21,16 @@ class MappingManager {
   void SetMappings(std::vector<Mapping> mappings);
   void AppendMapping(Mapping mapping);
 
+  [[nodiscard]] std::optional<std::filesystem::path> MapToFirstMatchingTarget(
+      const std::filesystem::path& source_path) const {
+    return orbit_source_paths_mapping::MapToFirstMatchingTarget(mappings_, source_path);
+  }
+
+  [[nodiscard]] std::optional<std::filesystem::path> MapToFirstExistingTarget(
+      const std::filesystem::path& source_path) const {
+    return orbit_source_paths_mapping::MapToFirstExistingTarget(mappings_, source_path);
+  }
```

Comment:
Maybe we can inline functions orbit_source_paths_mapping::MapToFirstMatchingTarget and orbit_source_paths_mapping::MapToFirstExistingTarget instead of calling them?

Reply:
What do you mean by inline? Do you mean copy the implementation in here? I explicilty opted for this design, to make the actual algorithm decoupled from the persistence class. This allows both the be tested independently.

Reply:
Yes, I meant to move implementation from src/SourcePathsMapping/Mapping.cpp here. But if it's worse for testing, than it doesn't make sense.